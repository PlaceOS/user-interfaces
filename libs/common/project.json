{
    "name": "common",
    "$schema": "../../node_modules/nx/schemas/project-schema.json",
    "projectType": "library",
    "sourceRoot": "libs/common/src",
    "prefix": "placeos",
    "targets": {
        "lint": {
            "executor": "@nx/eslint:lint",
            "outputs": ["{options.outputFile}"]
        },
        "test": {
            "executor": "@nx/jest:jest",
            "outputs": ["{workspaceRoot}/coverage/libs/common"],
            "options": {
                "jestConfig": "libs/common/jest.config.ts"
            }
        }
    },
    "tags": []
}
