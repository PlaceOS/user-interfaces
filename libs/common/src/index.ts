export * from './lib/common';
export * from './lib/uploads';
export * from './lib/feature-available.guard';
export * from './lib/remote-logging.service';
export * from './lib/timezone-helpers';
export * from './lib/user-idle-time.service';
export * from './lib/booking-rules';
export * from './lib/keep-alive.service';
export * from './lib/uploads.service';
export * from './lib/shorten.fn';
