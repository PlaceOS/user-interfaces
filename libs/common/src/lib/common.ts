export { BaseClass } from './base.class';
export { GoogleAnalyticsService } from './google-analytics.service';
export { HotkeysService } from './hotkeys.service';
export { SettingsService } from './settings.service';
export { VorlonService } from './vorlon.service';
export { VERSION } from './version';

export * from './angular-animations';
export * from './api';
export * from './application';
export * from './formatting';
export * from './general';
export * from './notifications';
export * from './placeos';
export * from './room-configuration.interface';
export * from './settings';
export * from './types';
export * from './user-state';
export * from './spec-helpers';
