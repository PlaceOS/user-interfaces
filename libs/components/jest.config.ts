/* eslint-disable */
export default {
    displayName: 'components',
    preset: '../../jest.preset.js',
    setupFilesAfterEnv: ['<rootDir>/src/test-setup.ts'],
    globals: {},
    coverageDirectory: '../../coverage/libs/components',
    snapshotSerializers: [
        'jest-preset-angular/build/serializers/no-ng-attributes',
        'jest-preset-angular/build/serializers/ng-snapshot',
        'jest-preset-angular/build/serializers/html-comment',
    ],
    transform: {
        '^.+.(ts|mjs|js|html)$': [
            'jest-preset-angular',
            {
                stringifyContentPathRegex: '\\.(html|svg)$',

                tsconfig: '<rootDir>/tsconfig.spec.json',
            },
        ],
    },
    transformIgnorePatterns: ['node_modules/(?!.*.mjs$)'],
};
