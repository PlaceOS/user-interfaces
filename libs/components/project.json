{
    "name": "components",
    "$schema": "../../node_modules/nx/schemas/project-schema.json",
    "projectType": "library",
    "sourceRoot": "libs/components/src",
    "prefix": "placeos",
    "targets": {
        "lint": {
            "executor": "@nx/eslint:lint",
            "outputs": ["{options.outputFile}"]
        },
        "test": {
            "executor": "@nx/jest:jest",
            "outputs": ["{workspaceRoot}/coverage/libs/components"],
            "options": {
                "jestConfig": "libs/components/jest.config.ts"
            }
        }
    },
    "tags": []
}
