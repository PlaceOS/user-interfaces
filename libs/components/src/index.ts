export * from './lib/components.module';

export { AuthorisedAdminGuard } from './lib/guards/authorised-admin.guard';
export { AuthorisedUserGuard } from './lib/guards/authorised-user.guard';
