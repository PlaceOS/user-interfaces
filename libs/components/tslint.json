{
    "extends": "../../tslint.json",
    "rules": {
        "directive-selector": [
            true,
            "attribute",
            "userInterfaces",
            "camelCase"
        ],
        "component-selector": [true, "element", "user-interfaces", "kebab-case"]
    },
    "linterOptions": {
        "exclude": ["!**/*"]
    }
}
