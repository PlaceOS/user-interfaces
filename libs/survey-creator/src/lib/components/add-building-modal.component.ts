import { Component, OnInit } from '@angular/core';
import { MatDialog, MatDialogRef } from '@angular/material/dialog';

@Component({
    selector: 'add-building-modal',
    template: `
        <section>
            <div class="header">
                <span class="dialog-title">Add Building</span>
                <span class="close" (click)="closeDialog()">
                    <mat-icon
                        aria-hidden="false"
                        aria-label="Material icon for closing dialog"
                        class="icon"
                        >close</mat-icon
                    >
                </span>
            </div>
            <main>
                <span class="small-title">Building image</span>
                <div class="upload-container"></div>
                <span class="small-title">Building name</span>

                <span class="small-title">Building location</span>
            </main>
            <footer>
                <button
                    mat-button
                    class="cancel-button"
                    color="basic"
                    (click)="closeDialog()"
                >
                    Cancel
                </button>
                <button
                    mat-button
                    class="add-button"
                    color="primary"
                    (click)="addBuilding()"
                >
                    Add
                </button>
            </footer>
        </section>
    `,
    styles: [
        `
            .header {
                display: flex;
                align-items: center;
                justify-content: space-between;
                background-color: #fff;
                height: 70px;
                border: 1px solid rgba(0, 0, 0, 0.12);
            }
            main {
                margin: 20px;
                display: flex;
                flex-direction: column;
            }
            .dialog-title {
                display: flex;
                font-size: 20px;
                line-height: 30px;
                font-weight: 800;
                padding: 20px;
            }
            .small-title {
                font-size: 14px;
                line-height: 24px;
                font-weight: 700;
                margin: 8px 0px 8px 0px;
            }
            .upload-container {
                background-color: #f8f8fa;
                width: 380px;
                height: 259px;
                border: 1px solid #d4d4d4;
                border-radius: 6px;
            }
            .cancel-button {
                display: flex;
                color: #292f5b;
                background-color: #fff;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
            }
            .add-button {
                display: flex;
                color: #fff;
                background-color: #292f5b;
                border-radius: 2px;
                margin: 20px 0px 20px 20px;
            }
            footer {
                display: flex;
                flex-direction: row;
                justify-content: flex-end;
                margin-right: 20px;
            }
            .close {
                display: flex;
                z-index: 99;
                text-align: right;
                vertical-align: middle;
                line-height: 12px;
                margin-right: 20px;
            }
        `,
    ],
})
export class AddBuildingModalComponent implements OnInit {
    constructor(public dialogRef: MatDialogRef<AddBuildingModalComponent>) {}

    ngOnInit(): void {}

    closeDialog() {
        this.dialogRef.close();
    }

    addAnotherBuilding(): void {
        return;
    }

    addBuilding(): void {
        return;
    }
}
