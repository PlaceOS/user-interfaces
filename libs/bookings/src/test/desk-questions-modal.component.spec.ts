import { createComponentFactory, Spectator } from '@ngneat/spectator/jest';
import { MatRadioModule } from '@angular/material/radio';
import { ReactiveFormsModule } from '@angular/forms';
import { MockComponent } from 'ng-mocks';
import { IconComponent } from '@placeos/components';

import { DeskQuestionsModalComponent } from '../lib/desk-questions-modal.component';

jest.mock('@placeos/common');

import * as common_mod from '@placeos/common';

describe('DeskQuestionsModalComponent', () => {
    let spectator: Spectator<DeskQuestionsModalComponent>;
    const createComponent = createComponentFactory({
        component: DeskQuestionsModalComponent,
        declarations: [MockComponent(IconComponent)],
        imports: [MatRadioModule, ReactiveFormsModule],
    });

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect(spectator.element).toMatchSnapshot();
    });

    it('should check for validation', (done) => {
        (common_mod as any).notifyError = jest.fn();
        spectator.component.event.subscribe(() => {
            expect(common_mod.notifyError).toHaveBeenCalledTimes(1);
            done();
        });
        spectator.click('footer button');
        expect(common_mod.notifyError).toHaveBeenCalledWith(
            'All the questions must be answered'
        );
        spectator.component.form.setValue({
            travelled: false,
            unwell: false,
            contact: false,
        });
        spectator.component.submit();
    });

    it('should show failure state', () => {
        expect('[failure]').not.toExist();
        spectator.component.form.setValue({
            travelled: false,
            unwell: true,
            contact: false,
        });
        spectator.component.submit();
        spectator.detectChanges();
        expect('[failure]').toExist();
        expect(spectator.element).toMatchSnapshot();
    });
});
