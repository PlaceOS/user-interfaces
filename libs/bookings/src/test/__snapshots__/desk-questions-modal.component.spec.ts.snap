// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeskQuestionsModalComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="relative"
  >
    <h2
      class="p-4 text-xl"
    >
      COVID-19 Questionnaire
    </h2>
    <main
      class="p-4 ng-untouched ng-pristine ng-invalid"
    >
      <div
        class="flex flex-col mb-4"
      >
        <label>
           Have you travelled overseas within the last 14 days?
          <span>
            *
          </span>
        </label>
        <mat-radio-group
          class="mat-radio-group space-x-2 ng-untouched ng-pristine ng-invalid"
          formcontrolname="travelled"
          role="radiogroup"
        >
          <mat-radio-button
            class="mat-radio-button _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-11-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-11-input"
                  name="mat-radio-group-9"
                  tabindex="0"
                  type="radio"
                  value="true"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                   
                </span>
                Yes
              </span>
            </label>
          </mat-radio-button>
          <mat-radio-button
            class="mat-radio-button _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-12-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-12-input"
                  name="mat-radio-group-9"
                  tabindex="0"
                  type="radio"
                  value="false"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                   
                </span>
                No
              </span>
            </label>
          </mat-radio-button>
        </mat-radio-group>
      </div>
      <div
        class="flex flex-col mb-4"
      >
        <label>
           Are you unwell or experiencing any cold or flu-like symptoms?
          <span>
            *
          </span>
        </label>
        <mat-radio-group
          class="mat-radio-group space-x-2 ng-untouched ng-pristine ng-invalid"
          formcontrolname="unwell"
          role="radiogroup"
        >
          <mat-radio-button
            class="mat-radio-button _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-14-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-14-input"
                  name="mat-radio-group-12"
                  tabindex="0"
                  type="radio"
                  value="true"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                   
                </span>
                Yes
              </span>
            </label>
          </mat-radio-button>
          <mat-radio-button
            class="mat-radio-button _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-15-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-15-input"
                  name="mat-radio-group-12"
                  tabindex="0"
                  type="radio"
                  value="false"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                   
                </span>
                No
              </span>
            </label>
          </mat-radio-button>
        </mat-radio-group>
      </div>
      <div
        class="flex flex-col"
      >
        <label>
           Have you had contact with anyone with suspected COVID-19?
          <span>
            *
          </span>
        </label>
        <mat-radio-group
          class="mat-radio-group space-x-2 ng-untouched ng-pristine ng-invalid"
          formcontrolname="contact"
          role="radiogroup"
        >
          <mat-radio-button
            class="mat-radio-button _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-17-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-17-input"
                  name="mat-radio-group-15"
                  tabindex="0"
                  type="radio"
                  value="true"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                   
                </span>
                Yes
              </span>
            </label>
          </mat-radio-button>
          <mat-radio-button
            class="mat-radio-button _mat-animation-noopable mat-accent"
          >
            <label
              class="mat-radio-label"
              for="mat-radio-18-input"
            >
              <span
                class="mat-radio-container"
              >
                <span
                  class="mat-radio-outer-circle"
                />
                <span
                  class="mat-radio-inner-circle"
                />
                <input
                  class="mat-radio-input cdk-visually-hidden"
                  id="mat-radio-18-input"
                  name="mat-radio-group-15"
                  tabindex="0"
                  type="radio"
                  value="false"
                />
                <span
                  class="mat-ripple mat-radio-ripple mat-focus-indicator"
                  mat-ripple=""
                >
                  <span
                    class="mat-ripple-element mat-radio-persistent-ripple"
                  />
                </span>
              </span>
              <span
                class="mat-radio-label-content"
              >
                <span
                  style="display: none;"
                >
                   
                </span>
                No
              </span>
            </label>
          </mat-radio-button>
        </mat-radio-group>
      </div>
    </main>
    <footer
      class="flex justify-center items-center p-2"
    >
      <button
        mat-button=""
      >
        Submit
      </button>
    </footer>
    <button
      close=""
      mat-dialog-close=""
      mat-icon-button=""
    >
      <i
        class="material-icons"
      >
        close
      </i>
    </button>
  </div>
  
  
</div>
`;

exports[`DeskQuestionsModalComponent should show failure state 1`] = `
<div
  id="root3"
>
  <main
    class="pt-8 relative"
    failure=""
  >
    <p
      class="p-4"
    >
       Your request to work from the office has been rejected based on your response to the compulsory Covid-19 questions. Please feel free to submit a new request when circumstances change in a way that changes your answer to the questions. 
    </p>
    <button
      close=""
      mat-dialog-close=""
      mat-icon-button=""
    >
      <i
        class="material-icons"
      >
        close
      </i>
    </button>
  </main>
  
  
</div>
`;
